### ISSUE TITLE
LastEvaluatedKey in QueryOutputTableTypeDef not treated as Optional contrary to boto3 documentation

### ISSUE THREAD

**Issue by Ricool06** [youtype/mypy_boto3_builder#324]
**Describe the bug**
LastEvaluatedKey in QueryOutputTableTypeDef is not treated as Optional by boto3-stubs despite it only being present if a DynamoDB query requires pagination. [The documentation for boto3 implies that it is optional here](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb/table/query.html#:~:text=If%20LastEvaluatedKey%20is%20present%20in%20the%20response%2C%20you%20will%20need%20to%20paginate%20the%20result%20set.).

Happy to submit a PR, but not 100% sure how the type builder works yet.

**To Reproduce**

1. Install `boto3-stubs[...]`
2. Run `mypy` on the following code sample

```python
import boto3
from mypy_boto3_dynamodb.type_defs import QueryOutputTableTypeDef

response: QueryOutputTableTypeDef = {
        "Items": [],
        "ConsumedCapacity": {},
        "ResponseMetadata": {
            "RequestId": "",
            "HTTPStatusCode": 200,
            "HTTPHeaders": {},
            "RetryAttempts": 0,
        },
        "Count": 0,
        "ScannedCount": 0,
        # "LastEvaluatedKey": {},
    }
```

**Actual output**

```
 error: Missing key "LastEvaluatedKey" for TypedDict "QueryOutputTableTypeDef"  [typeddict-item]
```

**Expected output**

```
Success: no issues found in 1 source file
```

**Additional context**
Ubuntu 20.04, installed with `poetry add boto3-stubs`, `boto3` version 1.34.146.

**Comment by vemel** [comment]
Hello. Thank you for the report.

Unfortunately, there is no way to tell which keys in output shapes are required or optional, because `required` array is not present in `botocore` shapes in 99% of cases. So, `builder` considers all output keys as required by default. 

However, there is a way to manually mark any key as optional here: https://github.com/youtype/mypy_boto3_builder/blob/main/mypy_boto3_builder/type_maps/required_attribute_map.py

So, I am going to add `ServiceNameCatalog.dynamodb -> ALL -> LastEvaluatedKey = False` in the next release. Let me know if you want to submit a PR.

**Comment by vemel** [comment]
Fixed in the latest release: https://github.com/youtype/mypy_boto3_builder/releases/tag/8.2.0

Pleas eudpate and let me know if it works as expected.

**Comment by vemel** [comment]
> > Fixed in the latest release: https://github.com/youtype/mypy_boto3_builder/releases/tag/8.2.0
> > Please update `mypy-boto3-dynamodb` and let me know if it works as expected.
> 
> Thanks, I wanted to submit a PR but this week has been busy.

Well, if you find any other keys that should be marked as `NotRequired` in output shapes - you can submit a PR.
